proto/dwtools/abase/l1.test/Str.test.s

```diff
@@ -8835,6 +8835,99 @@ ghij
 
 //
 
+function strLinesNearestReport( test )
+{
+  test.case = 'one letter to the left';
+  var src = 'function add( x,y ) { return x + y }';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 1, 7 ], gray : 1, numberOfLines : 1 });
+  var expected =  [ 'f', 'unctio', 'n add( x,y ) { return x + y }' ];
+  test.identical( got.nearest, expected );
+
+  test.case = 'a few letters to the left';
+  var src = 'function add( x,y ) { return x + y }';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 5, 7 ], gray : 1, numberOfLines : 1 });
+  var expected = [ 'funct', 'io', 'n add( x,y ) { return x + y }' ];
+  test.identical( got.nearest, expected );
+
+  test.case = 'one letter in the range';
+  var src = 'function add( x,y ) { return x + y }';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 5, 6 ], gray : 1, numberOfLines : 1 });
+  var expected = [ 'funct', 'i', 'on add( x,y ) { return x + y }' ];
+  test.identical( got.nearest, expected );
+
+  test.case = 'a few letters in the range';
+  var src = 'function add( x,y ) { return x + y }';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 1 });
+  var expected = [ 'funct', 'ion', ' add( x,y ) { return x + y }' ];
+  test.identical( got.nearest, expected );
+
+  test.case = 'one letter to the right';
+  var src = 'a\nb\nc';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 1, 4 ], gray : 1, numberOfLines : 1 });
+  var expected = [ 'a', '\nb\n', 'c' ];
+  test.identical( got.nearest, expected );
+
+  test.case = 'a few letters to the right';
+  var src = 'a\nb\ncd';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 1, 4 ], gray : 1, numberOfLines : 1 });
+  var expected = [ 'a', '\nb\n', 'cd' ];
+  test.identical( got.nearest, expected );
+
+  /* */
```

Comment:
Please, explain the principle of grouping of test cases. 

I can't understand why next two test cases is separated by `/* */`.

I can give one recommendation. Divide groups by changing of parameters.
First group - single line, the second - multiline. This groups can be separate test routines. 

Next step, change only parameter `numberOfLines` - other parameters should have constant value,
After that, change parameter `charsRangeLeft` and other parameters should have constant value.

It's example of such test groups
https://github.com/Wandalen/wTools/blob/master/proto/dwtools/abase/l1.test/Str.test.s#L7647